File: frontend/src/api/permission.ts
import api from './axios'

export interface PermissionResponse {
  id: number
  name: string
  code: string
  resource: string
  action: string
  description?: string
}

export const permissionApi = {
  getAllPermissions: () => api.get<PermissionResponse[]>('/permissions'),
  getPermissionsByResource: (resource: string) => api.get<PermissionResponse[]>(`/permissions/resource/${resource}`),
}

File: burndown-backend-ms/auth-service/src/main/java/com/burndown/auth/entity/Permission.java
package com.burndown.auth.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "permissions", schema = "ms_auth")
public class Permission {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(unique = true, nullable = false, length = 100)
    private String code;

    @Column(length = 50)
    private String resource;

    @Column(length = 50)
    private String action;

    @Column(length = 200)
    private String description;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}
